File: 4.server/src/main/java/com/shaolonger/monitorplatform/log/entity/HttpErrorLog.java
package com.shaolonger.monitorplatform.log.entity;

import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.Date;

@Entity
@Table(name = "lms_http_error_log")
@Data
public class HttpErrorLog {
    /**
     * 主键
     */
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "native")
    private Long id;

    /**
     * 项目标识
     */
    @NotEmpty(message = "projectIdentifier不能为空")
    private String projectIdentifier;

    /**
     * 日志类型
     * JS_ERROR、HTTP_ERROR、RESOURCE_LOAD、CUSTOM_ERROR
     */
    @NotEmpty(message = "logType不能为空")
    private String logType;

    /**
     * 创建时间
     */
    private Date createTime;

    /**
     * 用户ID
     */
    @NotNull(message = "userId不能为空")
    private Long userId;

    /**
     * 用户名
     */
    private String userName;

    /**
     * 页面URL
     */
    @Column(columnDefinition = "TEXT")
    private String pageUrl;

    /**
     * 页面关键字
     */
    private String pageKey;

    /**
     * 设备名称
     */
    private String deviceName;

    /**
     * 操作系统
     */
    private String os;

    /**
     * 浏览器名称
     */
    private String browserName;

    /**
     * 浏览器版本
     */
    private String browserVersion;

    /**
     * IP地址
     */
    private String ipAddress;

    /**
     * Http类型
     * 如"request"、"response"
     */
    private String httpType;

    /**
     * 完整的Http地址
     */
    @NotEmpty(message = "httpUrlComplete不能为空")
    @Column(columnDefinition = "TEXT")
    private String httpUrlComplete;

    /**
     * 缩写的Http地址
     */
    @Column(columnDefinition = "TEXT")
    private String httpUrlShort;

    /**
     * Http请求状态
     */
    @NotEmpty(message = "status不能为空")
    private String status;

    /**
     * Http请求状态文字描述
     */
    private String statusText;

    /**
     * 响应时长
     */
    private String resTime;
}

File: web-monitor-server/web-monitor-server-golang/model/validation/http_error_log.go
package validation

type AddHttpErrorLog struct {
	LogType           string `form:"logType"`
	ProjectIdentifier string `form:"projectIdentifier"`
	Cuuid             string `form:"cUuid"`
	Buid              uint64 `form:"bUid"`
	Buname            string `form:"bUname"`
	PageUrl           string `form:"pageUrl"`
	PageKey           string `form:"pageKey"`
	DeviceName        string `form:"deviceName"`
	Os                string `form:"os"`
	OsVersion         string `form:"osVersion"`
	BrowserName       string `form:"browserName"`
	BrowserVersion    string `form:"browserVersion"`
	IpAddress         string `form:"ipAddress"`
	Address           string `form:"address"`
	NetType           string `form:"netType"`
	HttpType          string `form:"httpType"`
	HttpUrlComplete   string `form:"httpUrlComplete"`
	HttpUrlShort      string `form:"httpUrlShort"`
	Status            string `form:"status"`
	StatusText        string `form:"statusText"`
	ResTime           string `form:"resTime"`
}

type GetHttpErrorLog struct {
	PageInfo
	StartTime         string `form:"startTime" binding:"omitempty,datetime=2006-01-02 15:04:05"`
	EndTime           string `form:"endTime" binding:"omitempty,datetime=2006-01-02 15:04:05"`
	LogType           string `form:"logType"`
	ProjectIdentifier string `form:"projectIdentifier"`
	Buname            string `form:"bUname"`
	PageUrl           string `form:"pageUrl"`
	HttpType          string `form:"httpType"`
	HttpUrlComplete   string `form:"httpUrlComplete"`
	HttpUrlShort      string `form:"httpUrlShort"`
	Status            string `form:"status"`
	StatusText        string `form:"statusText"`
}

type GetHttpErrorLogByGroup struct {
	PageInfo
	StartTime         string `form:"startTime" binding:"omitempty,datetime=2006-01-02 15:04:05"`
	EndTime           string `form:"endTime" binding:"omitempty,datetime=2006-01-02 15:04:05"`
	LogType           string `form:"logType"`
	ProjectIdentifier string `form:"projectIdentifier"`
	Buname            string `form:"bUname"`
	PageUrl           string `form:"pageUrl"`
	HttpType          string `form:"httpType"`
	HttpUrlComplete   string `form:"httpUrlComplete"`
	HttpUrlShort      string `form:"httpUrlShort"`
	Status            string `form:"status"`
	StatusText        string `form:"statusText"`
}

type GetLogCountByState struct {
	StartTime         string `form:"startTime" binding:"required,datetime=2006-01-02 15:04:05"`
	EndTime           string `form:"endTime" binding:"required,datetime=2006-01-02 15:04:05"`
	ProjectIdentifier string `form:"projectIdentifier" binding:"required"`
}
